services/statestorage/test/read_keys.go

```diff
@@ -0,0 +1,101 @@
+package test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Reading a Key", func() {
+	When("not providing a contract name", func() {
```

Comment:
I think those 3 tests:
not providing a contract name
providing a non existing contract
key doesn't exist
should be pushed into unit tests. 
Why? 
When you read those tests it seems for a moment that you are actually testing the driver and not the service itself. When diving down into it, I see that you actually test the service, but the abstraction of the driver is redundant in this case. 
I think that the general rule here is that if you have a test that is completely stateless, you should make it a unit test.